src/crypto/crypto.h

```diff
@@ -162,6 +163,20 @@ namespace crypto {
     return res;
   }
 
+  /* Generate a random value between 0 and range_max-1 in an unbiased way
+   */
+  template<typename T>
+  typename std::enable_if<std::is_pod<T>::value, T>::type rand_range(T range_max) {
+    typename std::remove_cv<T>::type res;
+    const typename std::remove_cv<T>::type max_value = (std::numeric_limits<T>::max() / range_max) * range_max;
+    do
+    {
+      res = rand<T>();
+    }
+    while (res >= max_value);
```

Comment:
Is the lower bound of the range implied 0 or implied `std::numeric_limits<T>::min()` ? This implementation is currently the former, but I'm not sure if allowing signed types is a good thing, it seems like the implementation could bust in those situations.